sender/conn_pools.go

```diff
@@ -1,11 +1,94 @@
 package sender
 
 import (
-	"github.com/open-falcon/transfer/g"
-	cpool "github.com/open-falcon/transfer/sender/conn_pool"
+	"errors"
+	"log"
+	"github.com/Cepave/transfer/g"
+	cpool "github.com/Cepave/transfer/sender/conn_pool"
 	nset "github.com/toolkits/container/set"
+	"strings"
 )
 
+var (
+	errInvalidDSNUnescaped = errors.New("Invalid DSN: Did you forget to escape a param value?")
+	errInvalidDSNAddr      = errors.New("Invalid DSN: Network Address not terminated (missing closing brace)")
+	errInvalidDSNNoSlash   = errors.New("Invalid DSN: Missing the slash separating the database name")
+)
+
+func parseDSN(dsn string) (cfg *cpool.InfluxdbConnection, err error) {
```

Comment:
Don't write another parse function. Please refer to `ParseConfig` in `g/cfg.go` and use `g.Config()` to access the configuration of influxdb. (e.g., `g.Config().influxdb.enabled`)